### PR TITLE
Fix for issue #4658: Add serialization performance tests for XmlSerializer and DCJS

### PR DESCRIPTION
**Pull request by SGuyGe**
@khdang @shmao @zhenlan This is the same as previous PR except multiple commits have been squashed into one.

### CODE REVIEW

File: src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                obj[i] = (byte)(i % 256);
+            }
+            return obj;
+        }
+
+        [Benchmark]
+        public void DeserializeByteArrayOf1K()
+        {
+            byte[] byteArrayOf1K = CreateByteArray(1024);
+            RunDcjsDeserializationTest(byteArrayOf1K, 10000);
```
Comment (stephentoub) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: You might consider doing this more concisely as a theory, e.g.
```C#
public static IEnumerable<object[]> DeserializationInputs()
{
    yield return new object[] { CreateByteArray(1024), 10000 };
    yield return new object[] { CreateByteArray(1024 * 1024), 10 };
    yield return new object[] { new string('a', 128), 10000 };
    yield return new object[] { new string('k', 1024 };
    ...
}

[Benchmark]
[MemberData("DeserializationInputs")]
public void RunDcjsDeserializationTest(object obj, int iterations)
{
    ...
}
```

File: src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+        private static List<int> CreateListOfInt(int count)
+        {
+            List<int> listOfInt = new List<int>();
+            for (int i = 0; i < count; i++)
+            {
+                listOfInt.Add(i);
+            }
+
+            return listOfInt;
```
Comment (stephentoub) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: Nit: could just be ```return Enumerable.Range(0, count).ToList();```

Comment (SGuyGe) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: Good suggestion, updated.

File: src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs (hunk 1 above)
Comment (SGuyGe) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: I think that would work for normal tests but probably not suited for perf tests.

Comment (stephentoub) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: Why?

Comment (SGuyGe) [inline src/System.Runtime.Serialization.Json/tests/Performance/DcjsDeserializationTests.cs]: The elimination of method name would make test results a bit harder to read. For example, DeserializeByteArrayOf1K would become System.Runtime.Tests.Performance.XmlSerializationTests.RunXmlSerializationTest(obj: [0, 1, 2, 3, 4, ...], iterations: 10000), and DeserializeStringOf128Byte would become System.Runtime.Tests.Performance.XmlSerializationTests.RunXmlSerializationTest(obj: "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"..., iterations: 10000).

**Comment by SGuyGe** [comment]
This is the same as PR #5361 which has been reviewed.

**Pull request merged**